mev/poll.go

```diff
@@ -22,22 +26,37 @@ func (p *Poller) run() {
 
 	bundles, err := p.client.GetBundles(context.Background(), &types.GetBundlesRequest{MinBlockHeight: uint64(lastHeight)})
 	if err != nil {
-		p.logger.Error("Error while querying SILO server for bundles", "err", err)
+		p.logger.Error("Error while querying bundle server", "err", err)
 		return
 	}
-	// TODO validate data, don't trust height
 	for height, bundles := range bundles.Bundles {
-		p.keeper.AddBundles(int64(height), bundles.Bundles)
+		p.keeper.SetBundles(int64(height), bundles.Bundles)
 	}
 
+	p.keeper.DropBundlesAtAndBelow(lastHeight - 1)
 }
 
 func NewPoller(ctx context.Context, logger log.Logger, config Config, keeper *Keeper, lastBlockProvider func() int64) (*Poller, error) {
 
 	logger.Info("Starting bundle provider poller")
 
-	// TODO security options?
-	grpcConn, err := grpc.DialContext(ctx, config.ServerAddr, grpc.WithInsecure())
+	if config.CertFile == "" && !config.Insecure {
+		return nil, fmt.Errorf("either certFile or insecure must be set")
+	}
+
+	var options []grpc.DialOption
+
+	if config.CertFile != "" {
+		creds, err := credentials.NewClientTLSFromFile(config.CertFile, "")
+		if err != nil {
+			return nil, fmt.Errorf("error while loading TLS certificate: %w", err)
+		}
+		options = append(options, grpc.WithTransportCredentials(creds))
+	} else {
+		options = append(options, grpc.WithInsecure())
+	}
+
+	grpcConn, err := grpc.DialContext(ctx, config.ServerAddr, options...)
```

Comment:
this is a silly nit, but you could just have 

```go 
var grpcOpt grpc.DialOption

if config.CertFile != "" {
   grpcOpt = grpc.WithTransportCredentials(creds)
else {
   grpcOpt = grpc.WithInsecure
}

conn, err := grpc.DialContext(ctx, config.ServerAddr, grpcOpt)
```

I don't think I care it's just always going to be a slice of one item and it's a variadic function anyway.

Reply:
Yes, good spot! Fixed.